diff --git a/compat/mingw.c b/compat/mingw.c
index be0c7a7a24..d7abdac8b2 100644
--- a/compat/mingw.c
+++ b/compat/mingw.c
@@ -1626,8 +1626,6 @@ static const char *quote_arg_msys2(const char *arg)
 
 static const char *parse_interpreter(const char *cmd)
 {
-	static char buf[MAX_PATH];
-	char *p, *opt;
 	ssize_t n; /* read() can return negative values */
 	int fd;
 
@@ -1639,25 +1637,8 @@ static const char *parse_interpreter(const char *cmd)
 	fd = open(cmd, O_RDONLY);
 	if (fd < 0)
 		return NULL;
-	n = read(fd, buf, sizeof(buf)-1);
 	close(fd);
-	if (n < 4)	/* at least '#!/x' and not error */
-		return NULL;
-
-	if (buf[0] != '#' || buf[1] != '!')
-		return NULL;
-	buf[n] = '\0';
-	p = buf + strcspn(buf, "\r\n");
-	if (!*p)
-		return NULL;
-
-	*p = '\0';
-	if (!(p = strrchr(buf+2, '/')) && !(p = strrchr(buf+2, '\\')))
-		return NULL;
-	/* strip options */
-	if ((opt = strchr(p+1, ' ')))
-		*opt = '\0';
-	return p+1;
+	return "env.exe";
 }
 
 /*
