diff --git a/contrib/buildsystems/CMakeLists.txt b/contrib/buildsystems/CMakeLists.txt
index 5b2c600c2d..a8441ed72f 100644
--- a/contrib/buildsystems/CMakeLists.txt
+++ b/contrib/buildsystems/CMakeLists.txt
@@ -970,7 +970,9 @@ endif()
 
 #to help with the install
 list(TRANSFORM git_shell_scripts PREPEND "${CMAKE_BINARY_DIR}/")
+list(TRANSFORM git_shell_scripts REPLACE "\.sh$" "")
 list(TRANSFORM git_perl_scripts PREPEND "${CMAKE_BINARY_DIR}/")
+list(TRANSFORM git_perl_scripts REPLACE "\.perl$" "")
 
 #install
 foreach(program ${PROGRAMS_BUILT})
