diff --git a/meson.build b/meson.build
index ed6133696d..96fb3aa674 100644
--- a/meson.build
+++ b/meson.build
@@ -1727,9 +1727,11 @@ if get_option('curl').enabled()
       dependencies: [libgit_curl],
     )
 
-    install_symlink(alias + executable_suffix,
-      install_dir: git_exec_path,
-      pointing_to: 'git-remote-http',
+    install_dir = '$MESON_INSTALL_DESTDIR_PREFIX/' + get_option('libexecdir') + '/git-core'
+    meson.add_install_script(
+      shell, '-c', 'ln ' +
+      install_dir + '/' + ('git-remote-http' + executable_suffix) + ' ' +
+      install_dir + '/' + (alias + executable_suffix),
     )
   endforeach
 endif
@@ -1747,9 +1749,11 @@ foreach alias : [ 'git-receive-pack', 'git-upload-archive', 'git-upload-pack' ]
     dependencies: [libgit_commonmain],
   )
 
-  install_symlink(alias + executable_suffix,
-    install_dir: git_exec_path,
-    pointing_to: 'git',
+  install_dir = '$MESON_INSTALL_DESTDIR_PREFIX/' + get_option('libexecdir') + '/git-core'
+  meson.add_install_script(
+    shell, '-c', 'ln ' +
+    install_dir + '/' + ('git' + executable_suffix) + ' ' +
+    install_dir + '/' + (alias + executable_suffix),
   )
 endforeach
 
@@ -1767,9 +1771,14 @@ foreach symlink : [
     pointing_to = '../libexec/git-core' / symlink
   endif
 
-  install_symlink(symlink,
-    install_dir: get_option('bindir'),
-    pointing_to: pointing_to,
+  install_dir = '$MESON_INSTALL_DESTDIR_PREFIX/' + get_option('bindir')
+  meson.add_install_script(
+    shell, '-c', 'mkdir -p ' + install_dir,
+  )
+  meson.add_install_script(
+    shell, '-c', 'ln ' +
+    install_dir + '/' + (pointing_to + executable_suffix) + ' ' +
+    install_dir + '/' + (symlink + executable_suffix),
   )
 endforeach
 
@@ -1870,9 +1879,11 @@ if perl_features_enabled
         pointing_to = '../libexec/git-core' / fs.stem(script)
       endif
 
-      install_symlink(fs.stem(script),
-        install_dir: get_option('bindir'),
-        pointing_to: pointing_to,
+      install_dir = '$MESON_INSTALL_DESTDIR_PREFIX/' + get_option('bindir')
+      meson.add_install_script(
+        shell, '-c', 'ln ' +
+        install_dir + '/' + pointing_to + ' ' +
+        install_dir + '/' + fs.stem(script),
       )
     endif
   endforeach
