diff --git a/meson.build b/meson.build
index ed6133696d..301091ed10 100644
--- a/meson.build
+++ b/meson.build
@@ -707,7 +707,7 @@ libgit_c_args = [
   '-DGIT_LOCALE_PATH="' + get_option('localedir') + '"',
   '-DGIT_MAN_PATH="' + get_option('mandir') + '"',
   '-DPAGER_ENV="' + get_option('pager_environment') + '"',
-  '-DSHELL_PATH="' + fs.as_posix(shell.full_path()) + '"',
+  '-DSHELL_PATH="/bin/sh"',
 ]
 libgit_include_directories = [ '.' ]
 libgit_dependencies = [ ]
@@ -1990,6 +1990,10 @@ foreach key, value : {
   build_options_config.set_quoted(key, value)
 endforeach
 
+build_options_config.set_quoted('PERL_PATH', '/usr/bin/perl')
+build_options_config.set_quoted('PYTHON_PATH', '/usr/bin/python')
+build_options_config.set_quoted('SHELL_PATH', '/bin/sh')
+
 configure_file(
   input: 'GIT-BUILD-OPTIONS.in',
   output: 'GIT-BUILD-OPTIONS',
