diff --git a/Documentation/meson.build b/Documentation/meson.build
index 138b03dae0..947c08bc53 100644
--- a/Documentation/meson.build
+++ b/Documentation/meson.build
@@ -322,7 +322,7 @@ foreach mode : [ 'diff', 'merge' ]
     command: [
       shell,
       '@INPUT@',
-      '..',
+      meson.project_source_root(),
       mode,
       '@OUTPUT@'
     ],
@@ -351,6 +351,7 @@ foreach manpage, category : manpages
     manpage_path = fs.stem(manpage) + '.' + category.to_string()
     manpage_target = custom_target(
       command: [
+        'env',
         xmlto,
         '-m', '@INPUT0@',
         '-m', '@INPUT1@',
